includes/ConvertToBlocks/Settings.php

```diff
@@ -0,0 +1,260 @@
+		</div>
+		<?php
+	}
+
+	/**
+	 * Registers section for the settings page.
+	 *
+	 * @return void
+	 */
+	public function register_section() {
+		add_settings_section(
+			$this->settings_section,
+			false,
+			false,
+			CONVERT_TO_BLOCKS_SLUG
+		);
+	}
+
+	/**
+	 * Registers setting fields.
+	 *
+	 * @return void
+	 */
+	public function register_fields() {
+		// Supported post types.
+		add_settings_field(
+			sprintf( '%s_post_types', CONVERT_TO_BLOCKS_SLUG ),
+			esc_html__( 'Supported Post Types', 'convert-to-blocks' ),
+			[ $this, 'field_post_types' ],
+			CONVERT_TO_BLOCKS_SLUG,
+			$this->settings_section,
+			[
+				'label_for' => sprintf( '%s_post_types', CONVERT_TO_BLOCKS_SLUG ),
+			]
+		);
+
+		register_setting(
+			$this->settings_group,
+			sprintf( '%s_post_types', CONVERT_TO_BLOCKS_SLUG ),
+			[
+				'sanitize_callback' => [ $this, 'sanitize_post_types' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders the `post_types` field.
+	 *
+	 * @return void
+	 */
+	public function field_post_types() {
+		$post_types  = get_option(
+			sprintf( '%s_post_types', CONVERT_TO_BLOCKS_SLUG ),
+			apply_filters( 'convert_to_blocks_default_post_types', CONVERT_TO_BLOCKS_DEFAULT_POST_TYPES )
+		);
+		$output_html = '';
+
+		foreach ( $this->post_types as $post_type ) {
+			$output_html .= sprintf(
+				'<label for="%1$s"><input name="%2$s[]" type="checkbox" %3$s id="%1$s" value="%4$s"> %5$s</label> <br>',
+				sprintf( '%s_post_types_%s', esc_attr( CONVERT_TO_BLOCKS_SLUG ), esc_attr( $post_type ) ),
+				sprintf( '%s_post_types', esc_attr( CONVERT_TO_BLOCKS_SLUG ) ),
+				checked( in_array( $post_type, $post_types, true ), 1, false ),
+				esc_attr( $post_type ),
+				esc_attr( ucfirst( $post_type ) )
+			);
+		}
+		?>
+		<fieldset>
+			<?php echo $output_html; // phpcs:ignore ?>
+		</fieldset>
+		<?php
+	}
+
+	/**
+	 * Sanitizes post_types values.
+	 *
+	 * @param array $input Array containing checked values.
+	 *
+	 * @return array Sanitized array.
+	 */
+	public function sanitize_post_types( $input ) {
+		if ( ! is_array( $input ) ) {
+			return [];
+		}
+
+		return array_map( 'sanitize_text_field', $input );
+	}
+
+	/**
+	 * Adds an admin notice if a filter is active for `post_type_supports_convert_to_blocks` as
+	 * this might overwrite the outcome of the settings stored in DB.
+	 */
+	public function filter_notice() {
+		if ( ! has_filter( 'post_type_supports_convert_to_blocks' ) ) {
+			return;
+		}
+
+		// URL to the settings panel.
+		$settings_url = add_query_arg(
+			[ 'page' => CONVERT_TO_BLOCKS_SLUG ],
+			admin_url( 'options-general.php' )
+		);
+		?>
+		<div class="notice notice-success is-dismissible">
+			<p>
+				<?php
+				printf(
+					/* translators: %1$s: link to switch to settings panel, %2$s: closing anchor tag */
+					esc_html__( 'A filter hook (post_type_supports_convert_to_blocks) is active which can lead to undesirable outcome. Kindly remove it and configure settings via the %1$sSettings Panel%2$s.', 'convert-to-blocks' ),
```

Comment:
> `Kindly remove it and configure settings via the %1$sSettings Panel%2$s.`

The notice is only displayed on the Convert to Blocks setting page, so we don't need this IMO.

Reply:
Also, a filter can be used for advanced conditions like preventing conversion for posts belonging to a category. So IMHO, we just need to notify users that one or more filters to `post_type_supports_convert_to_blocks` are being used, without any call to action.